src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/reference/Reference.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <Wizard
+    :ref="'wizard'"
+    :wizard="wizardStore.wizardData?.inviteType === PortalInviteType.WORK_EXPERIENCE ? workExperienceReferenceWizardConfig : characterReferenceWizardConfig"
+    :show-steps="false"
+  >
+    <template #header>
+      <v-container fluid class="bg-white">
+        <h2>{{ inviteType + " reference" }}</h2>
+        <div role="doc-subtitle">{{ `For applicant: ${applicantFirstName} ${applicantLastName}` }}</div>
+        <v-btn v-if="wizardStore.step !== 1" variant="text" rounded="lg" color="primary" @click="handleBack">{{ "< back" }}</v-btn>
+      </v-container>
+    </template>
+    <template #actions>
+      <v-container class="mb-8">
+        <v-row no-gutters>
+          <v-col>
+            <v-btn v-if="wizardStore.step === userReviewStep" rounded="lg" variant="flat" color="primary" @click="handleSubmit">Submit</v-btn>
+            <v-btn v-else rounded="lg" variant="flat" color="primary" @click="handleContinue">Continue</v-btn>
+          </v-col>
+        </v-row>
+      </v-container>
+    </template>
+  </Wizard>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import { useRoute } from "vue-router";
+
+import { getReference } from "@/api/reference";
+import characterReferenceWizardConfig from "@/config/character-reference-wizard";
+import workExperienceReferenceWizardConfig from "@/config/work-experience-reference-wizard";
+import { useAlertStore } from "@/store/alert";
+import { useWizardStore } from "@/store/wizard";
+import { PortalInviteType } from "@/utils/constant";
+
+import Wizard from "../Wizard.vue";
+
+export default defineComponent({
+  name: "Reference",
+  components: { Wizard },
+  async setup() {
+    const route = useRoute();
+    const { data } = await getReference(route.params.token as string);
+    const wizardStore = useWizardStore();
+    const alertStore = useAlertStore();
+    if (data?.portalInvitation?.inviteType === PortalInviteType.WORK_EXPERIENCE) {
+      wizardStore.initializeWizardForReference(workExperienceReferenceWizardConfig, data.portalInvitation);
+    } else if (data?.portalInvitation?.inviteType === PortalInviteType.CHARACTER) {
+      wizardStore.initializeWizardForReference(characterReferenceWizardConfig, data.portalInvitation);
+    }
+
+    return { alertStore, workExperienceReferenceWizardConfig, characterReferenceWizardConfig, wizardStore, PortalInviteType };
+  },
+  data() {
+    return {
+      inviteType: "",
+      applicantFirstName: "",
+      applicantLastName: "",
+    };
+  },
+  computed: {
+    userDeclinedStep(): number {
+      return this.wizardStore.steps.findIndex((step) => step.stage === "Decline") + 1;
+    },
+    userReviewStep(): number {
+      return this.wizardStore.steps.findIndex((step) => step.stage === "Review") + 1;
+    },
+  },
+  async mounted() {
+    this.inviteType = this.wizardStore.wizardData?.inviteType === PortalInviteType.WORK_EXPERIENCE ? "Work experience" : "Character";
```

Comment:
Same here, will these work as computed and remove from data and can get rid of mounted?

Reply:
moved this to a computed property for the title.